OpenEdXMobile/src/main/java/org/edx/mobile/view/custom/EdxRatingBar.java

```diff
@@ -39,5 +44,24 @@ private void init() {
         stars.getDrawable(0).setColorFilter(UNSELECTED_STAR_COLOR, PorterDuff.Mode.SRC_ATOP);
         stars.getDrawable(1).setColorFilter(SELECTED_STAR_COLOR_DARK, PorterDuff.Mode.SRC_ATOP);
         stars.getDrawable(2).setColorFilter(SELECTED_STAR_COLOR, PorterDuff.Mode.SRC_ATOP);
+        // Set accessibility label
+        setContentDescription(getResources().getString(R.string.rating_bar));
+    }
+
+    @Override
+    public void onInitializeAccessibilityEvent(AccessibilityEvent event) {
+        if (event.getEventType() == AccessibilityEvent.TYPE_VIEW_SELECTED) {
+            event.setSource(this);
+            event.setClassName(getClass().getName());
+            event.setPackageName(getContext().getPackageName());
+            event.setEnabled(isEnabled());
+            Map<String, CharSequence> map = new HashMap<>();
+            map.put("rating", String.valueOf((int) getRating()));
+            map.put("num_of_stars", String.valueOf(getNumStars()));
+            event.setContentDescription(ResourceUtil.getFormattedString(getResources(),
+                    R.string.rating_bar_selection, map));
+            return;
+        }
+        super.onInitializeAccessibilityEvent(event);
```

Comment:
We can take benefit of the `super` call in this case to populate the `event` object instead of applying the properties other than `contentDescription` ourselves. Following code would be a better as compared to the code above:
```java
        super.onInitializeAccessibilityEvent(event);
        if (event.getEventType() == AccessibilityEvent.TYPE_VIEW_SELECTED) {
            final Map<String, CharSequence> map = new HashMap<>();
            map.put("rating", String.valueOf((int) getRating()));
            map.put("num_of_stars", String.valueOf(getNumStars()));
            event.setContentDescription(ResourceUtil.getFormattedString(getResources(),
                    R.string.rating_bar_selection, map));
        }
```